Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -98,6 +103,16 @@ void doDepositionShapeN(const GetParticlePosition& GetPosition,
     amrex::ParallelFor(
         np_to_depose,
         [=] AMREX_GPU_DEVICE (long ip) {
+#if (defined AMREX_USE_GPU)
+            // To store present thread's time for deposition
+            __shared__ amrex::Real cost_shared;
+            cost_shared = 0.;
+
+            // By-hand timer: start
+            unsigned wt;
+            if (cost && load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::GpuClock) { wt = clock(); }
```

Comment:
Offline discussed: we could try to make these sections reusable in the following way.

- create an object that owns the shared memory
- in the constructor of the object, pass a bool that contains the activation based on the result of `cost && load_balance_costs_update_algo == LoadBalanceCostsUpdateAlgo::GpuClock`
- this in a local register variable
- if true, perform the clock and store in shared memory
- when destructed on running out-of-scope, check the cached bool's value and finalize the measurement if active

Reply:
I've added in `/Source/Utils` a smaller kernel timer class.  Maybe it actually belongs in AMReX (@kngott )?